tool/src/main.rs

```diff
@@ -296,11 +296,105 @@ unsafe fn fan_set(index: u8, duty: u8) -> Result<(), Error> {
     ec.fan_set(index, duty)
 }
 
+unsafe fn config_dump() -> Result<(), Error> {
+    iopl();
+
+    let mut ec = Ec::new(
+        StdTimeout::new(Duration::new(1, 0)),
+    )?;
+
+    let mut i = 0;
+    loop {
+        match ec.config_get_name(i) {
+            Err(err) => {
+                if 0 == i {
+                    // No configuration entries found.
+                    return Err(err);
+                }
+
+                // At least one entry found.
+                return Ok(());
+            },
+            Ok(name) => {
+                let desc = ec.config_get_desc(i)?;
+                let value = ec.config_get_value(i)?;
+                println!("{}: {}", name, desc);
+                println!("{}", value);
+            }
+        };
+        i += 1;
+    }
+}
+
+
+unsafe fn config_dump_name(name: String) -> Result<(), Error> {
+    iopl();
+
+    let mut ec = Ec::new(
+        StdTimeout::new(Duration::new(1, 0)),
+    )?;
+
+    // Find config entry with name
+    let mut i = 0;
+    loop {
+        match ec.config_get_name(i) {
+            Err(err) => return Err(err),
+            Ok(read_name) => {
+                if name == read_name {
+                    let desc = ec.config_get_desc(i)?;
+                    let value = ec.config_get_value(i)?;
+                    println!("{}: {}", name, desc);
+                    println!("{}", value);
+                    return Ok(());
+                }
+            },
+        }
+
+        i += 1;
+    }
+}
+
+
+unsafe fn config_set_value(name: String, value: i32) -> Result<(), Error> {
+    iopl();
+
+    let mut ec = Ec::new(
+        StdTimeout::new(Duration::new(1, 0)),
+    )?;
+
+    // Find config entry with name
+    let mut i = 0;
+    loop {
+        match ec.config_get_name(i) {
+            Err(err) => return Err(err),
+            Ok(read_name) => {
+                if name == read_name {
+                    return ec.config_set_value(i, value);
+                }
+            },
+        }
+
+        i += 1;
+    }
+}
+
+unsafe fn config_compact() -> Result<(), Error> {
+    iopl();
+
+    let mut ec = Ec::new(
+        StdTimeout::new(Duration::new(1, 0)),
+    )?;
+
+    ec.config_compact()
+}
+
 fn usage() {
     eprintln!("  console");
     eprintln!("  flash [file]");
     eprintln!("  flash_backup [file]");
     eprintln!("  fan [index] <duty>");
+    eprintln!("  config [name] [value]");
```

Comment:
If optional arguments are in [], then flash, flash_backup, fan, and print need to be adjusted.

Reply:
Looks like I misread this originally. I've fixed it to match fan with <> for optional arguments.